src/platform/plugins/shared/workflows_management/server/workflows_management/lib/create_index.ts

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+import type { ElasticsearchClient, Logger } from '@kbn/core/server';
+import type { MappingTypeMapping } from '@elastic/elasticsearch/lib/api/types';
+
+interface CreateIndexOptions {
+  esClient: ElasticsearchClient;
+  indexName: string;
+  mappings: MappingTypeMapping;
+  logger: Logger;
+}
+
+export const createIndexWithMappings = async ({
+  esClient,
+  indexName,
+  mappings,
+  logger,
+}: CreateIndexOptions): Promise<void> => {
+  try {
+    // Check if index already exists
+    const indexExists = await esClient.indices.exists({
+      index: indexName,
+    });
+
+    if (indexExists) {
+      logger.debug(`Index ${indexName} already exists`);
+      return;
+    }
+
+    logger.debug(`Creating index ${indexName} with mappings`);
+
+    // Create the index with proper mappings
+    await esClient.indices.create({
+      index: indexName,
+      mappings,
+      settings: {
```

Comment:
if you are changing the defaults of the index settings, maybe make sense to add some comments on why these settings

Reply:
removing, i think it's unnecessary